src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -962,8 +965,22 @@ private InstructorFeedbackResultsQuestionTable buildQuestionTableAndResponseRows
                 break;
             }
             
+            // If question specific sorting is not needed, responses are sorted
+            // by alphabetical order of giver's name
             if (questionDetails.isQuestionSpecificSortingRequired()) {
                 Collections.sort(responseRows, questionDetails.getResponseRowsSortOrder());
+            } else {
+                Collections.sort(responseRows, new Comparator<InstructorFeedbackResultsResponseRow>() {
+                    @Override
+                    public int compare(InstructorFeedbackResultsResponseRow a1,
+                            InstructorFeedbackResultsResponseRow a2) {
+                        return ComparisonChain.start()
+                                .compare(a1.getGiverTeam(), a2.getGiverTeam())
+                                .compare(a1.getGiverDisplayableIdentifier(),
+                                        a2.getGiverDisplayableIdentifier())
+                                .result();
+                    }
+                });
             }
             
         }
```

Comment:
Note how the `if` branch is much neater than the `else` branch. Extract the sorting to a separate method (e.g. `sortAlphabetically(...)`, put it where it belongs (usually, as a static method in the class of objects being sorted) and unit test that method separately. Things like sorting are independent small functionalities that should be implemented separately and unit tested separately (i.e. don't use integration/system tests to test such small standalone functionalities).